File: tests/generics/equatable-call.swift
public func equal<T: Equatable>(lhs: T, rhs: T) -> Bool {
	return lhs == rhs
}

public func match<T: Equatable>(lhs: T, rhs: T) -> Bool {
	return lhs ~= rhs
}

public func matchInts(lhs: Int, rhs: Int) -> Bool {
	return lhs ~= rhs
}

public func matchOptionals(lhs: Bool?, rhs: Bool?) -> Bool {
	return lhs ~= rhs
}

File: tests/generics/equatable-call.js
export function equal$lhs$rhs$(T, lhs, rhs) {
  return T.Equatable.$equals$(T, lhs, rhs);
}
export function match$lhs$rhs$(T, lhs, rhs) {
  return T.Equatable.$match$(T, lhs, rhs);
}
export function matchInts$lhs$rhs$(lhs, rhs) {
  return lhs === rhs;
}
export function matchOptionals$lhs$rhs$(lhs, rhs) {
  return lhs === rhs;
}